tests/pytests/test_vecsim_svs.py

```diff
@@ -25,24 +62,537 @@ def test_small_window_size():
+    no_index_proc_memory = get_redis_memory_in_mb(env)
+
+    # create index and measure memory. Expect it to increase by at least the size in bytes of vectors
+    create_vector_index(env, dim, alg='SVS-vamana')
+    waitForIndex(env, DEFAULT_INDEX_NAME)
+    env.assertEqual(get_tiered_debug_info(env, DEFAULT_INDEX_NAME, DEFAULT_FIELD_NAME)['INDEX_SIZE'], num_docs)
+    env.assertGreater(get_tiered_backend_debug_info(env, DEFAULT_INDEX_NAME, DEFAULT_FIELD_NAME)['INDEX_SIZE'], 0)
+
+    proc_memory = get_redis_memory_in_mb(env)
+    vectors_mem_mb = (num_docs * dim * 4) / (1024 * 1024)
+    env.assertGreater(proc_memory, no_index_proc_memory + vectors_mem_mb,
+                    message=f"no_index_proc_memory: {no_index_proc_memory}, proc_memory: {proc_memory}, vectors_mem_mb: {vectors_mem_mb}")
+
+    # drop index and measure memory. Expect it to decrease by at least the size in bytes of vectors
+    env.expect('FT.DROPINDEX', DEFAULT_INDEX_NAME).ok()
+    memory_after_drop = get_redis_memory_in_mb(env)
+    env.assertGreaterEqual(proc_memory - memory_after_drop, vectors_mem_mb, message=f"proc_memory: {proc_memory}, memory_after_drop: {memory_after_drop}, vectors_mem_mb: {vectors_mem_mb}")
+
+    # No operations on a dropped index are allowed
+    env.expect('FT.INFO', DEFAULT_INDEX_NAME).error().contains(f"no such index")
+    query = create_random_np_array_typed(dim, 'FLOAT32')
+    env.expect('FT.SEARCH', 'idx', f'*=>[KNN 1 @{DEFAULT_FIELD_NAME} $vec_param]', 'PARAMS', 2, 'vec_param',
+               query.tobytes(), 'NOCONTENT').error().contains(f"No such index")
+
+@skip(cluster=True)
+def test_drop_index_during_query():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2 WORKERS 1')
+    dim = 2
+    data_type = 'FLOAT32'
+    training_threshold = DEFAULT_BLOCK_SIZE
+    set_up_database_with_vectors(env, dim, num_docs=training_threshold, index_name=DEFAULT_INDEX_NAME, datatype=data_type, alg='SVS-VAMANA')
+    wait_for_background_indexing(env, DEFAULT_INDEX_NAME, DEFAULT_FIELD_NAME)
+    env.assertEqual(index_info(env, DEFAULT_INDEX_NAME)['num_docs'], training_threshold)
+
+    query = create_random_np_array_typed(dim, 'FLOAT32')
+    query_cmd = ['FT.SEARCH', DEFAULT_INDEX_NAME, f'*=>[KNN 10 @{DEFAULT_FIELD_NAME} $vec_param]', 'PARAMS', 2, 'vec_param', query.tobytes(), 'NOCONTENT']
+    # Build threads
+    t_query = threading.Thread(
+        target=runDebugQueryCommandPauseBeforeRPAfterN,
+        args=(env,query_cmd, 'Metrics Applier', 2),
+        daemon=True
+    )
+
+    # Start the query and the pause-check in parallel
+    t_query.start()
+
+    while getIsRPPaused(env) != 1:
+        time.sleep(0.1)
+
+    # drop the index while query is running
+    env.expect('FT.DROPINDEX', DEFAULT_INDEX_NAME).ok()
+
+    env.expect('FT.INFO', DEFAULT_INDEX_NAME).error().contains(f"no such index")
+    env.expect(*query_cmd).error().contains(f"No such index")
+    # Resume the query
+    setPauseRPResume(env)
+    t_query.join()
```

Comment:
Can / Should we test that query output is as expected (that is, the index was not dropped before execution has completed)?